nautobot/docs/development/model-features.md

```diff
@@ -0,0 +1,40 @@
+# Registry Model Features
+
+A dictionary of particular features (e.g. custom fields) mapped to the Nautobot models which support them, arranged by app.
+
+## Add a new feature to the `model_features` registry
+
+The `populate_model_features_registry()` function updates the registry model features with new apps. This is done by defining a list of dictionaries called `lookup_confs`. Each dictionary in `lookup_confs` contains the following three keys:
+
+- `feature_name`: The name of the feature to be updated in the registry.
+- `field_names`: A list of names of fields that must be present in order for the model to be considered a valid `model_feature`.
+- `field_attributes`: An optional dictionary of attributes to filter the fields by. Only model which fields match all the attributes specified in the dictionary will be considered. This parameter can be useful to narrow down the search for fields that match certain criteria. For example, if `field_attributes` is set to `{"related_model": RelationshipAssociation}`, only fields attr with `related_model` of `RelationshipAssociation` will be considered.
+
+To add a new feature to the `lookup_confs` list, follow these steps:
+
+1. Determine the name of the feature to be added.
```

Comment:
Should this be a `slug` rather than a `name`? I don't think there's anything *preventing* you from using a string with whitespace, capital letters, etc., but by convention we're using `snake_case` for all existing features.